src/QsCompiler/SyntaxProcessor/ScopeTools.fs

```diff
@@ -204,9 +204,18 @@ type SymbolTracker(globals : NamespaceManager, sourceFile, parent : QsQualifiedN
             let properties = (defaultLoc, InvalidIdentifier, ResolvedType.New InvalidType, false)
             properties |> LocalVariableDeclaration.New false, ImmutableArray<_>.Empty
 
-        let buildCallable kind fullName (decl : ResolvedSignature) attributes = 
-            let deprecated = attributes |> SymbolResolution.TryFindRedirect |> SymbolResolution.GenerateDeprecationWarning (fullName, qsSym.RangeOrDefault)
-            for msg in deprecated do msg |> addDiagnostic
+        let buildCallable kind fullName (decl : ResolvedSignature) attributes =
+            // if parent is deprecated, no longer generate warning
+            let parentAttrs = 
+                match GlobalSymbols().TryGetCallable parent (parent.Namespace, sourceFile) with 
+                | Found decl -> decl.Attributes
+                | _ -> ArgumentException "the given NamespaceManager does not contain a callable with the given parent name" |> raise
+            let deprecated =
+                if not (Seq.exists BuiltIn.MarksDeprecation parentAttrs) then Some (attributes |> SymbolResolution.TryFindRedirect |> SymbolResolution.GenerateDeprecationWarning (fullName, qsSym.RangeOrDefault))
+                else None
+            match deprecated with
+            | Some dep -> for msg in dep do msg |> addDiagnostic
+            | _ -> ()
```

Comment:
You could also put the call to `addDiagnostic` inside the `if` expression - I think that would be simpler since you could avoid pattern matching on the intermediate `deprecated` variable.